src/generate.rs

```diff
@@ -356,16 +355,71 @@ fn register_block(registers: &[Register], defs: &Defaults) -> Result<Tokens> {
     let mut i = 0;
     // offset from the base address, in bytes
     let mut offset = 0;
-    for register in util::expand(registers) {
-        let pad = if let Some(pad) = register.offset.checked_sub(offset) {
+    let mut registers_expanded = vec![];
+
+    for register in registers {
+        match *register {
+            Register::Single(ref _into) => registers_expanded.push(register.clone()),
+            Register::Array(ref _into, ref array_info) => {
+                let index_convertible = if let Some(ref indexes) = array_info.dim_index {
```

Comment:
Could you add a comment about what this does? It wasn't entirely clear from reading the contents. Is it checking that `indexes == 0..info.dim`? In that case this may be simpler to write this as `indexes.iter().eq(0..info.dim)`.